src/storage/mod.rs

```diff
@@ -488,15 +488,35 @@ impl<E: Engine> TestStorageBuilder<E> {
     }
 }
 
+/// `Storage` implements transactional KV APIs and raw KV APIs on a given `Engine`. An `Engine`
+/// provides low level KV functionality. `Engine` has multiple implementations. When a TiKV server
+/// is running, a `RaftKv` will be the underlying `Engine` of `Storage`. The other two types of
+/// engines are for test purpose.
+///
+/// `Storage` is actually a reference to storage. It is cloneable, and a clone of a `Storage` is
+/// actually a reference to the same storage, and is equivalent to the original one. It will be
+/// automatically stopped and destroyed when it and all its clones are dropped.
+///
+/// All read and write methods are performed on **a single region**, which is specified in the
```

Comment:
May be not suitable to mention `Region` here, because for other kind of `Engine` there is no Region concept.
How about: Notice that read and write methods may not be performed over full data in most cases, i.e. when underlying engine is `RaftKv`, which limits data access in the range of a single region according to specified `ctx` parameter.